docs/release/release-process.md

```diff
@@ -0,0 +1,293 @@
+
+At this point, the `HEAD` commit in the release branch will be our release candidate.  The build
+pipeline will automatically be started due to the create branch event, so we can start to perform
+testing on this build.  Note that it should be the exact same as what is currently in `master` since
+they are using the same commit and have the same tag.  Also note that this is not the official
+release build since we have not made the official release tag yet (e.g. `v0.5.0`).
+
+The `master` branch can now be opened for new features since we have a safe release branch to
+continue bug fixes and improvements for the release itself.  Essentially, `master` is free to now
+diverge from the release branch.
+
+#### Bug Fixes in Release Branch
+
+During our testing of the release candidate, we may find issues or bugs that we triage and decide we
+want to fix before the release goes out. In order to fix a bug in the release branch, the following
+process is recommended:
+
+1. Make the bug fix into `master` first through the normal PR process
+    1. If the applicable code has already been removed from `master` then simply fix the bug
+       directly in the release branch by opening a PR directly against the release branch
+1. Backport the fix by performing a cherry-pick of the fix's commit hash (**not** the merge commit)
+   from `master` into the release branch.  For example, to backport a fix from master to `v0.5.0`,
+   something like the following should be used:
+
+    ```console
+    git fetch --all
+    git checkout -b release-0.5 upstream/master
+    git cherry-pick -x <fix commit hash>
+    ```
+
+1. Open a PR with the cherry-pick commit targeting the release-branch
+
+After all bugs have been fixed and backported to the release branch, we can move on to releasing the
+stacks that the Crossplane community owns. Note that core Crossplane has still not been released
+yet.
+
+### Draft Release Notes
+
+We're getting close to starting the official release, so you should take this opportunity to draft
+up the release notes. You can create a [new release draft
+here](https://github.com/crossplaneio/crossplane/releases/new).  Make sure you select "This is a
+pre-release" and hit "Save draft" when you are ready to share and collect feedback.  Do **not** hit
+"Publish release" yet.
+
+You can see and follow the template and structure from [previous
+releases](https://github.com/crossplaneio/crossplane/releases).
+
+### Stack Release Process
+
+At this point, we should have a release branch created in all repos with all issues resolved that
+were triaged to be release blockers.  This section will walk through how to release the Stacks
+themselves and does not directly apply to core Crossplane.
+
+In the **release branch** for each Stack, you should update the version tag or metadata in:
+
+* `app.yaml` - `version`
+* `install.yaml` - `image`
+* `integration_tests.sh` - `STACK_IMAGE`
```

Comment:
Ah see that it is addressed here. I am fine to leave it out above as anyone running the release will hopefully have read this part as well.